src/IB/IBFEMethod.cpp

```diff
@@ -318,6 +319,27 @@ assemble_poisson(EquationSystems& es, const std::string& /*system_name*/)
     }
 }
 
+std::string
+get_data_time_str(const double data_time, const double current_time, const double half_time, const double new_time)
```

Comment:
Is it worth making this more general?

Reply:
It seems fine to me - those are the time points we expect everywhere in the library.

Reply:
It would be nice to, at some point, move this switch statement into a more general place but we can handle that in another PR.

Reply:
I could move this to `ibtk_utilities.h`.

Reply:
Let's add that to this PR.

Reply:
or just merge - up to you.

Reply:
I will move it and squash after you update tests.

Reply:
Since `half_time = 0.5*(current_time + new_time)`, I thought it might make sense just to have folks pass in `current_time` and `new_time`:
```
template <class T>
inline std::string
get_data_time_str(const T data_time, const T current_time, const T new_time)
{
    const T half_time = static_cast<T>(0.5) * (current_time + new_time);
    if (SAMRAI::tbox::MathUtilities<T>::equalEps(data_time, current_time))
    {
        return "current";
    }
    else if (SAMRAI::tbox::MathUtilities<T>::equalEps(data_time, half_time))
    {
        return "half";
    }
    else if (SAMRAI::tbox::MathUtilities<T>::equalEps(data_time, new_time))
    {
        return "new";
    }
    else
    {
        return "unknown";
    }
}
```
OK?